app/endpoints/advert.py

```diff
@@ -0,0 +1,403 @@
+        raise HTTPException(
+            status_code=400,
+            detail="Invalid id, group_manager_id must be a valid group id",
+        )
+
+    try:
+        db_advertiser = models_advert.Advertiser(
+            id=str(uuid.uuid4()),
+            name=advertiser.name,
+            group_manager_id=advertiser.group_manager_id,
+        )
+
+        return await cruds_advert.create_advertiser(db_advertiser=db_advertiser, db=db)
+    except ValueError as error:
+        raise HTTPException(status_code=422, detail=str(error))
+
+
+@router.delete(
+    "/advert/advertisers/{advertiser_id}",
+    status_code=204,
+    tags=[Tags.advert],
+)
+async def delete_advertiser(
+    advertiser_id: str,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.admin)),
+):
+    """
+    Delete an advertiser. All adverts associated with the advertiser will also be deleted from the database.
+
+    **This endpoint is only usable by administrators**
+    """
+    advertiser = await cruds_advert.get_advertiser_by_id(
+        advertiser_id=advertiser_id, db=db
+    )
+    if advertiser is None:
+        raise HTTPException(
+            status_code=404,
+            detail="Invalid advertiser_id",
+        )
+
+    for advert in advertiser.adverts:
+        await cruds_advert.delete_advert(advert_id=advert.id, db=db)
+
+    await cruds_advert.delete_advertiser(advertiser_id=advertiser_id, db=db)
+
+
+@router.patch(
+    "/advert/advertisers/{advertiser_id}",
+    status_code=204,
+    tags=[Tags.advert],
+)
+async def update_advertiser(
+    advertiser_id: str,
+    advertiser_update: schemas_advert.AdvertiserUpdate,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.admin)),
+):
+    """
+    Update an advertiser
+
+    **This endpoint is only usable by administrators**
+    """
+    advertiser = await cruds_advert.get_advertiser_by_id(
+        advertiser_id=advertiser_id, db=db
+    )
+    if not advertiser:
+        raise HTTPException(
+            status_code=404,
+            detail="Invalid advertiser_id",
+        )
+
+    await cruds_advert.update_advertiser(
+        advertiser_id=advertiser_id, advertiser_update=advertiser_update, db=db
+    )
+
+
+@router.get(
+    "/advert/me/advertisers",
+    response_model=list[schemas_advert.AdvertiserComplete],
+    status_code=200,
+    tags=[Tags.advert],
+)
+async def get_current_user_advertisers(
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user_a_member),
+):
+    """
+    Return all advertisers the current user can manage.
+
+    **The user must be authenticated to use this endpoint**
+    """
+
+    user_advertisers: list[models_advert.Advertiser] = []
+
+    existing_advertisers: Sequence[
+        models_advert.Advertiser
+    ] = await cruds_advert.get_advertisers(db=db)
+
+    for advertiser in existing_advertisers:
```

Comment:
Could not this be in cruds with a simple query like `select(models_advert.Advertiser).where(models_advert.Advertiser.group_manager_id.in_(user_groups))` ?